lib/inc/internal/ruby/ruby_value.hpp

```diff
@@ -79,12 +81,28 @@ namespace facter { namespace ruby {
          */
         leatherman::ruby::VALUE value() const;
 
+        /**
+         * Exposes an owned ruby VALUE as a facter ruby_value
+         * @param child the owned child object to wrap
+         * @param key the query string for the child
+         * @return pointer to the ruby_value wrapper for the child object
+         */
+        ruby_value const* wrap_child(leatherman::ruby::VALUE child, std::string key) const;
+
+        /**
+         * Get a cached ruby_value wrapper for a child VALUE
+         * @param key the query string for the child
+         * @return pointer to the ruby_value wrapper, or nullptr if none exists
+         */
+        ruby_value const* child(const std::string& key) const;
+
      private:
         static void to_json(leatherman::ruby::api const& ruby, leatherman::ruby::VALUE value, facts::json_allocator& allocator, facts::json_value& json);
         static void write(leatherman::ruby::api const& ruby, leatherman::ruby::VALUE value, std::ostream& os, bool quoted, unsigned int level);
         static void write(leatherman::ruby::api const& ruby, leatherman::ruby::VALUE value, YAML::Emitter& emitter);
 
         leatherman::ruby::VALUE _value;
+        mutable std::unordered_map<std::string, std::unique_ptr<ruby_value>> _children;
```

Comment:
I'd like a comment here about why mutable, and what changes might require that we rethink this approach.